sympy/polys/monomials.py

```diff
@@ -274,6 +274,49 @@ def monomial_gcd(A, B):
     """
     return tuple([ min(a, b) for a, b in zip(A, B) ])
 
+def monomial_ngcd(monomials):
+    """
+    Computes the greatest common divisor (GCD) of the exponents for each
+    variable in the monomials.
+
+    Parameters
+    ==========
+
+    monomials : iterable of tuples
+        Represents monomials, where each monomial is a tuple of exponents for variables.
+
+    Returns
+    =======
+
+    tuple
+        The greatest common divisor of the exponents for each variable in the
+        monomials.
+
+    Examples
+    ========
+
+    >>> from sympy.polys.monomials import monomial_ngcd
+    >>> monomial_ngcd([(1, 0), (0, 2), (2, 0), (0, 1)])
+    (0, 0)
+
+    """
+    zero_monom = tuple(0 for _ in monomials[0])
+
+    # If the first monomial is already the zero monomial, return early
+    if monomials[0] == zero_monom:
+        return zero_monom
```

Comment:
We might have the `zero_monom` as not the first tuple so a quick check here could be:
```
if zero_monom in monomials:
    return zero_monom
```